src/paths.rs

```diff
@@ -0,0 +1,91 @@
+use app_dirs::{AppDataType, AppInfo};
+use std::path::PathBuf;
+
+pub const LOCK_FILE_NAME: &str = "requirements.lock";
+
+const APP_INFO: AppInfo = AppInfo {
+    name: "dmenv",
+    author: "Tanker",
+};
+
+use crate::error::*;
+
+pub struct Paths {
+    pub project: PathBuf,
+    pub venv: PathBuf,
+    pub lock: PathBuf,
+    pub setup_py: PathBuf,
+}
+
+pub struct PathsResolver {
+    pub venv_outside_project: bool,
+    pub python_version: String,
+
+    pub project_path: PathBuf,
+}
+
+impl PathsResolver {
+    pub fn new(project_path: PathBuf, python_version: &str) -> Self {
+        PathsResolver {
+            venv_outside_project: false,
+            project_path,
+            python_version: python_version.into(),
```

Comment:
Nit: It's a bit of an inconsistent API to take a PathBuf instead of &Path, and then a &str instead of String.
Make up your mind, do you want owned objects or refs =] ?
(I'd just take a String, I think)